csrc/multidevice/communication.h

```diff
@@ -102,28 +87,27 @@ class Communication : public Expr {
   }
 
   const DeviceMesh& senderMesh() const {
```

Comment:
I suggest to remove the methods `senderMesh` and `receiverMesh` and to inline them in the code when needed